src/Hng.Web/Controllers/ProductController.cs

```diff
@@ -35,6 +35,21 @@ public async Task<ActionResult<ProductDto>> CreateProduct([FromBody] ProductCrea
             return Ok(successResponse);
         }
 
+        [HttpPost("add-products")]
+        [Authorize]
+        [ProducesResponseType(typeof(ProductDto), StatusCodes.Status201Created)]
+        public async Task<ActionResult<ProductsDto>> AddProducts([FromBody] AddMultipleProductDto body)
+        {
+            var loggedInUserId = HttpContext.User.FindFirst(ClaimTypes.Sid).Value;
```

Comment:
There is a service that already does this. Check the AuthenticationService in Infrastructure. You should move whatever logic you want to perform with this to the handler. Keep this controller simple.